tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -62,13 +65,42 @@ def get_plugin_apps(self):
 
   def is_active(self):
     """The scalars plugin is active iff any run has at least one scalar tag."""
+    if self._db_connection_provider:
+      # The plugin is active if one relevant tag can be found in the database.
+      db = self._db_connection_provider()
+      cursor = db.execute(
+          'SELECT 1 FROM Tags WHERE Tags.plugin_name = ? LIMIT 1',
```

Comment:
This code will work, except in the garbage data edge case. If you want to make this strictly correct right now, `SELECT run_id FROM ...` in a loop, and `return True` once you've verified the associated `Run` also exists.

Reply:
Strictly speaking, if by "garbage data" you mean "anything that vacuum.cc would vacuum" then in order to determine if a tag is garbage or not we have to check if the tag has a run, if the run has an experiment, and then if the experiment has a user.  Is that what you have in mind for "verified the associated Run also exists"?  It seems a bit heavyweight to do on every lookup, but I suppose maybe we could cache the run_ids of verified runs somewhere to make it cheaper.

Reply:
Hmm, legit question: How important is being exactly correct about the plugin being active? Also, what are the performance repercussions of being correct? Is seems like we'd have to make a few queries?

If a relevant row is garbage data, then I think that means that at some point, it was valid data, which meant that the scalars plugin was active at some point. Incorrectly (and temporarily) deeming the plugin to be active in that case is ... not that bad IMO - and might be worth it if it speeds up the `is_active` method. Furthermore, the vacuuming logic will I believe clear the garbage row in due time anyway.

Reply:
Sounds great. It seems like the consensus here per Justine is correct as particular situation requires, which makes sense to me. In this case, I think this query meets that bar - the `is_active` method is mainly for determining whether to place the plugin name within the navigation bar, and if any scalar data had been previously detected, we can deem the plugin to be active.